### PR TITLE
Move dev dependencies to Gemfile and lock `rubocop` version

### PR DESCRIPTION
**Pull request by DannyBen**
As discussed in #89, this PR offers a little housekeeping:

- Move dev dependencies from `gemspec` to `Gemfile`
- Add `Gemfile.lock` to `.gitignore`
- Lock `rubycop` version to the version that worked the last time someone used it, so that it is not blocking CI tests.

### CODE REVIEW

**Comment by DannyBen** [comment]
I see that tests are passing (except Ruby 3.0, as expected).
Once this is merged, we can proceed with #89 and hopefully #91.

File: Gemfile [hunk 1]
```diff
@@ -1,2 +1,12 @@
 source 'https://rubygems.org'
+
+gem 'jeweler'
+gem 'pry'
+gem 'rake'
+gem 'rdoc'
+gem 'rspec'
```
Comment (kenpratt) [inline Gemfile:7]: What do you think about putting major versions here (eg `gem 'rspec', '~> 3.0')`? Not necessary?

**Review by kenpratt — APPROVED** [review]
Thanks for noticing & fixing this.

File: Gemfile (hunk 1 above)
Comment (DannyBen) [reply Gemfile:7]: I my opinion, absolutely not necessary. In [all my gems](http://github.dannyben.com/ruby/) I am using the same approach. We want devs to always use the latest. This way, you make it a habbit to occasionally do `bundle update` and `bundle outdated`, to ensure everything is new.

The only reason to occasionally lock versions, is when something is broken and we do not have time to fix it.

I am no expert on these matters - but this is how I roll.

Comment (kenpratt) [reply Gemfile:7]: Got it, that sounds good to me.

**Pull request merged**